### ISSUE TITLE
getCurrentPosition does not work with gps only

### ISSUE THREAD

**Issue by Libero793** [Baseflow/flutter-geolocator#123]
## 🐛 Bug Report

getCurrentPosition only works when GPS ist set manually to energysafing/highAccuracy Mode on Android. If the Location Mode is set To GPS only, getCurrentPosition doesnt respond and the Code Stops waiting for a signal without any reply

### Expected behavior

working with GPS data Mode "Only Smartphone" 
### Reproduction steps

You Will never get the Response print "print(position)" in this example, if gps isnt set to energysaving or high accuracy manually

`Geolocator geolocator = Geolocator()..forceAndroidLocationManager = true;
    GeolocationStatus geolocationStatus  = await Geolocator().checkGeolocationPermissionStatus();
    print(geolocationStatus);
    Position position = await Geolocator().getCurrentPosition(desiredAccuracy: LocationAccuracy.lowest);
    print(position);`

### Configuration

**Version:** 1.x

**Platform:** 
- [ ] :iphone: iOS
- [ x ] :robot: Android

**Comment by diogoribeiro23** [comment]
@Libero793 @mvanbeusekom I found the issue. if you have only gps active you cant be indoors for it to work (OBVIOUSLY :rofl:). It works fine!

**Comment by mvanbeusekom** [comment]
@diogoribeiro23  thanks for the feedback, indeed when you are inside the phone probably cannot get a fix from the GPS satellite. 

@Libero793 can you confirm this would also solve the problem on you side (in other words can you go outside and move around a bit, to see if you start getting GPS updates :P )?

**Comment by Libero793** [comment]
Hey there,

moving around doesnt fix it for me. It would be great if there is an option to choose which ohne of the 3 gps services you want to enable

**Comment by ibrierley** [comment]
I've been looking at a problem very similar to this for a wee while, and had found this thread, so posting what I had found as it may help others...

Depending upon the version of Android, you may need one of the following permissions as well...

```
<!--<uses-permission android:name="android.hardware.location.gps" />-->
<uses-permission android:name="android.hardware.location" />
```

See this thread [https://stackoverflow.com/questions/39603270/when-do-i-need-android-hardware-location-gps-and-android-hardware-location-netwo](url)

If others can confirm, it may be worth adding to the docs (maybe even have a check as part of the library?), as it took me a long time to figure this out.

**Comment by Libero793** [comment]
My Device (Galaxy S8) still only turns gps back to the mode it was set before, and the GPS Data only option still doesnt do the job.

**Comment by Libero793** [comment]
So it actually seems like the behaviour described by @diogoribeiro23 is correct - the gps "phone only" option does not work indoor, this is why it searchs forever for gps. Still it would be usefull if you could trigger a specific mode or at least prevent the gps for searching in the limited "phone gps only" mode

**Comment by Libero793** [comment]
@mvanbeusekom  would be nice if we could get a fix for this, would help me massively with my current project

**Comment by mvanbeusekom** [comment]
@Libero793 we are open for pull-requests. At the moment I am personally pretty occupied with another project so I cannot really give any realistic indication on when I can get to it myself.

**Comment by jesaiasmaia** [comment]
Guys, any solution for this? my app is like Uber, so many users put in locations config "phone only" and there is no feedback from the plugin saying that was not possible get location. Is there some thing that we can get the location mode the user is using?

**Comment by rafaelterada** [comment]
Good Morning. I had this same problem, I made a change and it worked. In the LocationUpdatesUsingLocationManagerTask class, you get the following code:
locationManager.requestLocationUpdates (
                 mActiveProvider,
                 mLocationOptions.getTimeInterval (),
                 mLocationOptions.getDistanceFilter (),
                 this,
                 looper);
I changed to:
Criteria criteria = new Criteria ();
         criteria.setAccuracy (Criteria.ACCURACY_FINE);
         criteria.setAltitudeRequired (false);
         criteria.setCostAllowed (true); 
locationManager.requestLocationUpdates (mLocationOptions.getTimeInterval (),
                 mLocationOptions.getDistanceFilter (), criteria, this,
                 Looper.getMainLooper ());
And it worked. I do not know why I was not otherwise ...

**Comment by mvanbeusekom** [comment]
Fixed in version 6.0.0
